test_rclcpp/CMakeLists.txt

```diff
@@ -192,10 +192,31 @@ if(BUILD_TESTING)
       set(rmw_implementation2_is_fastrtps TRUE)
     endif()
 
+    set(rmw_implementation1_is_connext FALSE)
+    set(rmw_implementation2_is_connext FALSE)
+    if(rmw_implementation1 MATCHES "(.*)connext(.*)")
+      set(rmw_implementation1_is_connext TRUE)
+    endif()
+    if(rmw_implementation2 MATCHES "(.*)connext(.*)")
+      set(rmw_implementation2_is_connext TRUE)
+    endif()
+
+    set(rmw_implementation1_is_cyclonedds FALSE)
+    set(rmw_implementation2_is_cyclonedds FALSE)
+    if(rmw_implementation1 MATCHES "(.*)cyclonedds(.*)")
+      set(rmw_implementation1_is_cyclonedds TRUE)
+    endif()
+    if(rmw_implementation2 MATCHES "(.*)cyclonedds(.*)")
+      set(rmw_implementation2_is_cyclonedds TRUE)
+    endif()
+
     # Skipped tests between fastrtps and others, as the node names are communicated with a different mechanism.
     # TODO(ivanpauno): Reenable this tests after the other implementations also use one Participant per Context.
     if(
-      (NOT rmw_implementation1_is_fastrtps AND NOT rmw_implementation2_is_fastrtps) OR
+      (rmw_implementation1_is_fastrtps AND rmw_implementation2_is_cyclonedds) OR
+      (rmw_implementation1_is_cyclonedds AND rmw_implementation2_is_fastrtps) OR
+      (rmw_implementation1_is_cyclonedds AND rmw_implementation2_is_cyclonedds) OR
+      (rmw_implementation1_is_connext AND rmw_implementation2_is_connext) OR
```

Comment:
The new condition only conditionally enables known combinations. Therefore this test is not used for any other RMW impl. The logic should be inverted to mark the combinations know to not work as skipped - but enable the test for all other cases (including unknown combinations).

Reply:
> the test should be created unconditionally and have the SKIP_TEST flag passed in the cases which are intentionally skipped

Uses `SKIP_TEST` in e0842247b0b9ea3d2efad8858cbab738a54af8b1

> The logic should be inverted to mark the combinations know to not work as skipped - but enable the test for all other cases (including unknown combinations).

I was going to do this, but on second thought I think whitelisting fits better. Blacklisting would make sense if it were assumed all unknown rmw implementations can communicate with any other unknown implementation. However; I think the real assumption is that only DDS based rmw implementations should be able to communicate with each other. In that case a whitelist can enable DDS-based cross-vendor tests while avoiding enabling cross-vendor tests for implementations we don't have info about. For example, the whitelist avoids enabling cross vendor tests for `iceoryx` and `connext` without adding knowledge of `iceoryx`'s existance to `test_rclcpp`.

Reply:
We don't know all DDS-based RMW implementations and it would be nice if those can run these these tests without requiring them to be modified.

Reply:
> We don't know all DDS-based RMW implementations and it would be nice if those can run these these tests without requiring them to be modified.

Do we have any CLI or environment variable options for forcing skipped tests to run? That would be nice for those unknown non-DDS implementations that want to run system tests without the cross-vendor tests.

Reply:
I don't know if `ctest` / `pytest` offer such an option. (Not applicable here but `pytest` has an option to run tests marked with `xfail`.)

Reply:
I opened an issue to track this for now https://github.com/ros2/system_tests/issues/429

Reply:
I am pretty sure the time to write that ticket exceeds the necessary time to invert the logic in this PR...